cluster/store/logger.go

```diff
@@ -0,0 +1,179 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package store
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/sirupsen/logrus"
+)
+
+func NewHCLogrusLogger(name string, logger *logrus.Logger) hclog.Logger {
+	return &hclogLogrus{
+		entry: logrus.NewEntry(logger),
+		name:  fmt.Sprintf("%s ", name),
+	}
+}
+
+type hclogLogrus struct {
+	entry *logrus.Entry
+	name  string
+}
+
+func (hclogger *hclogLogrus) GetLevel() hclog.Level {
+	switch hclogger.entry.Logger.Level {
+	case logrus.TraceLevel:
+		return hclog.Trace
+	case logrus.DebugLevel:
+		return hclog.Debug
+	case logrus.InfoLevel:
+		return hclog.Info
+	case logrus.WarnLevel:
+		return hclog.Warn
+	case logrus.ErrorLevel:
+	case logrus.FatalLevel:
```

Comment:
question: I always assumed logging fatal would kill/panic the process. I feel like this change breaks that assumption. Therefore do we log Fatal anywhere and should update that code to panic/exit afterward ?

Reply:
actually i thin you'r right w.r.t https://github.com/sirupsen/logrus/blob/dd1b4c2e81afc5c255f216a722b012ed26be57df/hooks/syslog/syslog.go#L37C1-L39C32 

i may need to change fatal and panic to panic

Reply:
the problem is now that hclog doesn't have the same level :D

Reply:
well given that also that logrus<->gclog they don't have the same leveling and to [follow docs in hclog](https://github.com/hashicorp/go-hclog/blob/main/logger.go#L43)  it seems that is the max. we can do as level point of view, hclog won't panic that is the diff

Reply:
That sounds OK to me